cmd/wharf/config.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"os"
+
+	"github.com/iver-wharf/wharf-cmd/pkg/aggregator"
+	"github.com/iver-wharf/wharf-cmd/pkg/provisioner"
+	"github.com/iver-wharf/wharf-cmd/pkg/provisionerapi"
+	"github.com/iver-wharf/wharf-cmd/pkg/worker"
+	"github.com/iver-wharf/wharf-core/v2/pkg/config"
+)
+
+// Config holds all configurable settings for wharf-api.
+//
+// The config is read in the following order:
+//
+// 1. File: ~/.config/iver-wharf/wharf-cmd/wharf-cmd-config.yml
+//
+// 2. File: ./wharf-cmd-config.yml
+//
+// 3. File from environment variable: WHARF_CMD_CONFIG
+//
+// 4. Environment variables, prefixed with WHARF_CMD
+//
+// Each inner struct is represented as a deeper field in the different
+// configurations. For YAML they represent deeper nested maps. For environment
+// variables they are joined together by underscores.
+//
+// All environment variables must be uppercased, while YAML files are
+// case-insensitive. Keeping camelCasing in YAML config files is recommended
+// for consistency.
+type Config struct {
+	Worker         worker.Config
```

Comment:
Suggest moving this to its own package.

The documentation is not shown in pkg.go.dev on `main` packages :(

Moving it to `pkg/config` should be fine, and then you can reuse the types everywhere instead of having the config types be scattered.